transformer_engine/pytorch/ops/basic/cast_float8.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+"""Fusible operation for identity."""
+
+from __future__ import annotations
+from typing import Optional
+
+import torch
+
+from ...fp8 import FP8GlobalStateManager, get_fp8_te_dtype
+from ...tensor import Float8Tensor
+from ..op import BasicOperation, OperationContext
+from .._common import is_float8_tensor
+
+
+class CastFloat8(BasicOperation):
```

Comment:
I believe you said that it is mostly an utility op for tests, right? We should probably mention that in this documentaiton.

Reply:
Also maybe we should consider generalizing it a bit with a name that is not specific to FP8? (Like just Quantize)?

Reply:
It could be a helpful op for users as well. For example, if a user wants to have discrete layers for design reasons but still wants to fuse some operations with FP8 casts:
```python
act = te.ops.Sequential(te.ops.GeLU(), te.ops.Quantize())
linear = te.ops.Sequential(te.ops.Linear())
y = act(x)
z = linear(y)
```